hugr/src/types/type_row.rs

```diff
@@ -40,6 +43,8 @@ impl TypeRow {
 
     #[inline(always)]
     /// Returns the port type given an offset. Returns `None` if the offset is out of bounds.
+    // Note/TODO: it might be good to disable this if we are indexing over (past) a Row Variable,
+    // as substitution could change where in the row the offset refers.
```

Comment:
I started in branch `feat/row_typevar4` (!). I remember from last time that it was *easier* to change the variable-arity FunctionType (to FunTypeVarArgs) that the fixed-arity one (to Signature), even if the ideal is probably the other way around, so that's what I started doing. But it's still very hard: you need to change every `FunctionType::new` into either `FunTypeVarArgs::new` or `FunctionType::try_new....unwrap` (as FunctionType::new doesn't exist, it needs to check against row vars).

Maybe I've got it wrong and changing fixed-arity to Signature(::try_new) is the "easier" way, I dunno. I am minded to leave this and think about coming back to it as a separate/second PR ?

Reply:
See #1044 (not building yet)